### PR TITLE
Augment `eval+cache!` exception with form + location info

### PR DESCRIPTION
**Pull request by philomates**
## before

```
  :cause
  "Execution error (PSQLException) at org.postgresql.jdbc.PgPreparedStatement/setObject (PgPreparedStatement.java:1029).\nCan't infer the SQL type to use for an instance of java.util.Date. Use setObject() with an explicit Types value to specify the type to use.\n",
  :data
  ***:clojure.error/class org.postgresql.util.PSQLException,
   :clojure.error/line 1029,
   :clojure.error/cause
   "Can't infer the SQL type to use for an instance of java.util.Date. Use setObject() with an explicit Types value to specify the type to use.",
   :clojure.error/symbol
   org.postgresql.jdbc.PgPreparedStatement/setObject,
   :clojure.error/source "PgPreparedStatement.java",
   :clojure.error/phase :execution***,
  :phase :execution***

Execution error (ExceptionInfo) at nextjournal.clerk.eval/eval+cache! (eval.clj:151).
Execution error (PSQLException) at org.postgresql.jdbc.PgPreparedStatement/setObject (PgPreparedStatement.java:1029).
Can't infer the SQL type to use for an instance of java.util.Date. Use setObject() with an explicit Types value to specify the type to use.
```

## after

```
:cause
  "Execution error (PSQLException) at org.postgresql.jdbc.PgPreparedStatement/setObject (PgPreparedStatement.java:1029).\nCan't infer the SQL type to use for an instance of java.util.Date. Use setObject() with an explicit Types value to specify the type to use.\n",
  :data
  ***:clojure.error/class org.postgresql.util.PSQLException,
   :clojure.error/line 1029,
   :clojure.error/cause
   "Can't infer the SQL type to use for an instance of java.util.Date. Use setObject() with an explicit Types value to specify the type to use.",
   :clojure.error/symbol
   org.postgresql.jdbc.PgPreparedStatement/setObject,
   :clojure.error/source "PgPreparedStatement.java",
   :clojure.error/phase :execution,
   :clerk/form
   (sort-by
    second
    (frequencies
     (map
      :DLFREKDN
      (utils/services-in-date-range start-date end-date)))),
   :clerk/file "dev/insights/billing/an_invoice.clj"***,
  :phase :execution***

Execution error (ExceptionInfo) at nextjournal.clerk.eval/eval+cache! (eval.clj:151).
Execution error (PSQLException) at org.postgresql.jdbc.PgPreparedStatement/setObject (PgPreparedStatement.java:1029).
Can't infer the SQL type to use for an instance of java.util.Date. Use setObject() with an explicit Types value to specify the type to use.
```

### CODE REVIEW

File: src/nextjournal/clerk/eval.clj [hunk 1]
```diff
@@ -148,7 +148,10 @@
           (assoc :nextjournal/interned @!interned-vars))))
     (catch Throwable t
       (let [triaged (main/ex-triage (Throwable->map t))]
-        (throw (ex-info (main/ex-str triaged) triaged))))))
+        (throw (ex-info (main/ex-str triaged)
+                        (assoc triaged
+                               :clerk/form form
+                               :clerk/file file)))))))
```
Comment (philomates) [inline src/nextjournal/clerk/eval.clj:154]: maybe drop the `clerk` ns?

Comment (mk) [reply src/nextjournal/clerk/eval.clj:154]: Can we maybe reuse the `clojure.core/file` name like in https://github.com/nextjournal/clerk/blob/990749d3218b35765d8f9d8a826638e53f6d552d/src/nextjournal/clerk/analyzer.clj#L130-L132?

**Comment by philomates** [comment]
shall we merge this? This came in use in another instance today, so would be nice to have in `main`

**Comment by mk** [comment]
@philomates thanks for this and sorry for the slow turnaround.

**Pull request merged**